src/Ledger/Certs.lagda

```diff
@@ -71,7 +71,8 @@ cwitness (deregdrep c _)     = just c
 cwitness (ccreghot c _)      = just c
 \end{code}
 \begin{code}[hide]
-cwitness (reg _ _)           = nothing
+cwitness (reg _ zero)        = nothing
+cwitness (reg c (suc _))     = just c
```

Comment:
Is this reflected in the spec documentation (of a previous era, most likely)?  Or did you find this by conformance testing?  If the latter, is this another adjustment to make the spec fit the implementation?  If so, we should keep a log of such changes and document them somewhere.  I think this is important enough to justify marking my review "changes requested." ...but if you could add some text to the literate version of the module (i.e., in the `Certs.lagda` file), outside of a code block, then I'll approve.  Also, could we start putting an asterisk at the end of any line in the CHANGELOG that describes a change that moves the spec away from the original and/or closer to the implementation, so we can keep track of such changes?

Reply:
I recommend adding text in the `Certs.lagda` file---a sentence or two explaining what's in your remark above ("Actually it requires a witness only if the deposit fee is specified in the certificate. It has nothing to do with the type of the credential.").

Reply:
Andre suggested that we put anything related to the `reg` cert into a hidden code block, since it's sort of an implementation detail that would unnecessarily clutter up the spec. Do you think we should make all of this visible and explain it? I don't think it would make sense to explain the code in literate Agda if the code itself is not visible. I could instead add a regular comment which is only visible when reading the source.

Reply:
I guess the idea is to discourage people from using a certificate that is going to be deprecated in the next era.

Reply:
In general I favor transparency over obfuscation, but if there's a good reason `reg` should be hidden from the reader (e.g., it shouldn't be used in Conway), that's fine.  But, for maintainability, I agree that your comment above should at least appear in a code comment.